### PR TITLE
refactor: extract the cmd clients factory in a separate package and set up the CommonOptions before creating the commands

### PR DESCRIPTION
**Pull request by ccojocar**
#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

#### Description

- Extract the cmd clients factory in a separate package and set up the CommonOptions before creating the commands
- Make the common flags persistent for all commands
- Remove some short flags names which were conflicting with the short name of batch flag
- Update the description of common flags
- Move the PullSecrets option from common commands into the specific commands (create env and devpod) 

#### Which issue this PR fixes

part of #1016

fixes #

### CODE REVIEW

File: pkg/jx/cmd/create_env_test.go [hunk 1]
```diff
@@ -133,15 +133,15 @@ func TestCreateEnvRun(t *testing.T) {
 	helmValuesConfig.ExposeController.Config.HTTP = "false"
 	helmValuesConfig.ExposeController.Config.TLSAcme = "false"
 
+	commonOpts := cmd.NewCommonOptionsWithFactory(factory)
+	commonOpts.In = console.In
+	commonOpts.Out = console.Out
+	commonOpts.Err = console.Err
```
Comment (pmuir) [inline pkg/jx/cmd/create_env_test.go]: I wonder if we should do a helper func for this as it's quiet a common pattern in tests - we can add that later!

Comment (ccojocar) [reply pkg/jx/cmd/create_env_test.go]: Agree. Let's add it as a follow up.

**Pull request merged**